src/events/SDL_events.c

```diff
@@ -815,6 +831,14 @@ SDL_PumpEvents(void)
 #endif
 
     SDL_SendPendingSignalEvents();  /* in case we had a signal handler fire, etc. */
+
+    if (SDL_GetEventState(SDL_POLLSENTINEL) == SDL_ENABLE) {
```

Comment:
Previously the poll sentinel was only added if at least one other event was in the queue.
Should a new sentinel only be added if there are other events in the queue and/or if there are no other sentinels in the queue?

Reply:
No, for simplicity we always add a sentinel event. This guarantees behavior regardless of what other event actions are happening.